### PR TITLE
Watchlater likes functionality

### PR DESCRIPTION
**Pull request by dusariudaykumar**
# Feat: Added Like and watchLater Functionalities

## Added :
- Vertical video card Component
- Like and watchLater pages
- Context and Reducer
- API services for Like and watchLater

### CODE REVIEW

File: src/Contexts/data-context.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  const removeVideoFromWatchLaterHandler = async (videoId) => {
+    try {
+      const {
+        data: { watchlater },
+      } = await removeVideoFromWatchLater(encodedToken, videoId);
+      dataDispatch({ type: "REMOVE_FROM_WATCH_LATER", payload: watchlater });
+    } catch (error) {
+      console.log(error.message);
+    }
+  };
```
Comment (Chiragmodi01) [inline src/Contexts/data-context.js]: I'd prefer keeping this logic separate.

File: src/Pages/HomePage/HomePage.jsx [hunk 2]
```diff
@@ -1,15 +1,40 @@
...
+  const addVideoToWatchLaterHandler = async (video) => {
+    try {
+      const {
+        data: { watchlater },
+      } = await addVideoToWatchLater(encodedToken, video);
+      dataDispatch({ type: "ADD_TO_WATCH_LATER", payload: watchlater });
+    } catch (error) {
+      console.log(error.message);
+    }
+  };
```
Comment (Chiragmodi01) [inline src/Pages/HomePage/HomePage.jsx]: again the same, prefer extracting these logics.

File: src/Pages/SinglePageVideo/SinglePageVideo.jsx [hunk 3]
```diff
@@ -1,15 +1,40 @@
 import { useParams } from "react-router-dom";
-import { BiLike, BiDislike } from "react-icons/bi";
+import { HiThumbUp, HiOutlineThumbUp } from "react-icons/hi";
```
Comment (Chiragmodi01) [inline src/Pages/SinglePageVideo/SinglePageVideo.jsx:2]: you can export all icons from one index file, and import any of the icons from that.

**Review by sharmayatendra — APPROVED** [review]
code looks clean
design is consistent
file structure is good

**Review by Chiragmodi01 — COMMENTED** [review]
Added few sanity comments.
Functionality seems to be working fine.
UI also looks good.

**Pull request merged**